modules/python/pylib/syslogng/__init__.py

```diff
@@ -1,17 +1,18 @@
 #############################################################################
-# Copyright (c) 2015-2016 Balabit
+# Copyright (c) 2022 Balazs Scheidler <bazsi77@gmail.com>
 #
-# This program is free software; you can redistribute it and/or modify it
-# under the terms of the GNU General Public License version 2 as published
-# by the Free Software Foundation, or (at your option) any later version.
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
 #
-# This program is distributed in the hope that it will be useful,
+# This library is distributed in the hope that it will be useful,
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
 #
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
```

Comment:
Couple of notes:

1) I rewrote the entire file, the history shows it's changed compared to that file, but only because this is a file at the same location and I've squashed a rename into the git history. None of the code lines match the previous versions.
2) I don't think we want to enforce that all Python-based drivers have to be under the GPL, I was under the impression that that was not a goal. The Java code in a similar setting is also LGPL and in general syslog-ng as a project wanted to allow whole plugins to be licensed  differently than the core.
3) the code could be trivially bypassed (if someone wanted due to licensing reasons) by importing from _syslogng directly, but I don't want that to happen.
4) at the moment, it's not clearly articulated that the separation between syslog-ng C code and the Python implementation of a driver would constitute derived work or not. Normally, the source code compiled with gcc is not constituted derived work, even if there's actual linkage between gcc runtime code and the resulting binary. With the same logic, Python implemented code should not become derived work. With the new library I just wrote, and its licensing under the LGPL, this is more clearly articulated. And the python module implemented in C can remain GPLd, which choice I would actually prefer.
5) the code is trivial and thus probably not copyrightable

With all of the above said, 1) alone should be enough for me to license that file under a new license. The rest explains why I did that.

In any case, I am ok with adding the balabit copyright line there to avoid any other confusion down the line, I am doing that in the next iteration.

Reply:
I am leaving this open, even though I consider this resolved. lmk.

Reply:
Thank you for the extremely detailed clarification.
I can only agree :)